src/ubx-message.toit

```diff
@@ -200,90 +503,833 @@ class AckAck extends Message:
+      sv-used-mask        := 0b00000000_00001000
+      health-mask         := 0b00000000_00110000
+      diff-corr-mask      := 0b00000000_01000000
+      smoothed-mask       := 0b00000000_10000000
+      orbit-source-mask   := 0b00000111_00000000
+      eph-avail-mask      := 0b00001000_00000000
+      alm-avail-mask      := 0b00010000_00000000
+      ano-avail-mask      := 0b00100000_00000000
+      aop-avail-mask      := 0b01000000_00000000
+
+
+      quality      = (flags & quality-mask) >> quality-mask.count-trailing-zeros
+      health       = (flags & health-mask) >> health-mask.count-trailing-zeros
+      orbit-source = (flags & orbit-source-mask) >> orbit-source-mask.count-trailing-zeros
+      alm-avail    = (flags & alm-avail-mask) >> alm-avail-mask.count-trailing-zeros
+      eph-avail    = (flags & eph-avail-mask) >> eph-avail-mask.count-trailing-zeros
+      ano-avail    = (flags & ano-avail-mask) >> ano-avail-mask.count-trailing-zeros
+      aop-avail    = (flags & alm-avail-mask) >> alm-avail-mask.count-trailing-zeros
+      diff-corr    = ((flags & diff-corr-mask) >> diff-corr-mask.count-trailing-zeros) != 0
+      sv-used      = ((flags & sv-used-mask) >> sv-used-mask.count-trailing-zeros) != 0
+      smoothed     = ((flags & smoothed-mask) >> smoothed-mask.count-trailing-zeros) != 0
+
+      if (eph-avail != 0) or (alm-avail != 0) or (ano-avail != 0) or (aop-avail != 0):
+        orbit-info-avail = true
+
+    else if src-id == NavSvInfo.ID:
+      offset = index * 12
+      channel = LITTLE-ENDIAN.uint8 payload offset + 8
+
+      sv-id = LITTLE-ENDIAN.uint8 payload offset + 9
+      cno = LITTLE-ENDIAN.uint8 payload offset + 12
+      elev = LITTLE-ENDIAN.int8 payload offset + 13
+      azim = LITTLE-ENDIAN.int8 payload offset + 14
+      pr-res = (LITTLE-ENDIAN.uint32 payload offset + 16).to-float / 100 // Scaled in cm.
+      flags = LITTLE-ENDIAN.uint32 payload offset + 10
+
+      quality-mask   := 0b00000111
+      quality = (LITTLE-ENDIAN.uint32 payload offset + 11) & quality-mask
+
+      sv-used-mask     := 0b00000001
+      diff-corr-mask   := 0b00000010
+      orbit-avail-mask := 0b00000100
+      orbit-eph-mask   := 0b00001000
+      unhealthy-mask   := 0b00010000
+      orbit-alm-mask   := 0b00100000
+      orbit-aop-mask   := 0b01000000
+      smoothed-mask    := 0b10000000
+
+      // Directly usable
+      diff-corr        = ((flags & diff-corr-mask) >> diff-corr-mask.count-trailing-zeros) != 0
+      sv-used          = ((flags & sv-used-mask) >> sv-used-mask.count-trailing-zeros) != 0
+      orbit-info-avail = ((flags & orbit-avail-mask) >> orbit-avail-mask.count-trailing-zeros) != 0
+      smoothed         = ((flags & smoothed-mask) >> smoothed-mask.count-trailing-zeros) != 0
+
+      // Translated to return outputs matching the later definition
+      unhealthy-raw  := (flags & unhealthy-mask) >> unhealthy-mask.count-trailing-zeros
+      if unhealthy-mask == 1:
```

Comment:
The `mask` can't be 1 or 0. It's always equal to `0b00010000`.
I guess it should be `unhealthy-raw`.
Isn't this simply: `health = unhealthy-raw + 1` ?